lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -864,11 +864,10 @@ def create_autoscaling_group(connection, module):
                     )
 
         # check for attributes that aren't required for updating an existing ASG
-        desired_capacity = desired_capacity or as_group['DesiredCapacity']
-        min_size = min_size or as_group['MinSize']
-        max_size = max_size or as_group['MaxSize']
+        desired_capacity = desired_capacity if desired_capacity is not None else as_group['DesiredCapacity']
+        min_size = min_size if min_size is not None else as_group['MinSize']
```

Comment:
Moving the definition of `min_size` et al. from line 664 would be a lot more elegant:

```
min_size = module.params.get('min_size', as_group['MinSize'])
etc...
```

Reply:
This doesn't work as expected, unfortunately.
```
>>> test_dict = {'one': 1, 'two': None}
>>> print(test_dict.get('two'))
None
>>> print(test_dict.get('two', 2))
None
>>> print(test_dict.get('three', 3))
3
```

Reply:
have reverted as requested